### DISCUSSION TITLE
Random freezes of tracking

### DISCUSSION THREAD

**Discussion by ANDREW007CZ** [bailuk/AAT#144] [General]
I'm on Android 13, Sony Xperia 5 III.
AAT v 1.24

Randomly the tracking in background freezes. After opening app it looks like time goes on but length, altitude, speed and any other stuff get stuck.

I believe it is due to short outtage of GPS system and AAT does not send some new request to re-enable GPS system to work. I don't really understand how it works this is just wild guess.

I've encountered this problem twice in all 6 recordings I've recently did.

**Comment by ANDREW007CZ** [comment]
* all permisions are given and battery optimisation is not restricted at all

**Comment by bailuk** [comment]
Possible cause 1: Accuracy filter is set. You can disable  accuracy filter in `Preferences -> Activity X`
Possible cause 2:  There is a security feature in newer Android versions to prevent apps track your date without informing you. See issue #120

Technically there is no need to re-enable GPS when the signal is lost.

Does tracking continue after a while once AAT is in foreground again? Is the GPS icon in the Android status bar visible when this happens? What is the GPS Status display of AAT (... or on or off) ?

**Reply by sebicodes** [reply]
I had the same issue changing to a newer phone from Android 11 to Android 13. But on troubleshooting I tested what openTracks does ( https://github.com/OpenTracksApp/OpenTracks ), and this app tracks continuously without having the permission to access location "all the time" (but is telling me every 2-4 minutes that something is "becoming idle"), maybe that is interesting. I don't understand that behaviour.
Giving permission "always" to aat as mentioned in #120 solved my problem. Is it possible to ask for that permission directly on installing? That would be much easier...

**Comment by ANDREW007CZ** [comment]
With the latest version (1.25.2 from f-droid) the dropouts are far less often. I think it has something to do with loss of GPS signal. Once it happens AAT does not recover tracking and manual intervention is needed. For example opening map app does request the GPS location and as soon as is fixed, AAT starts tracking again.
Permission for location is set to "always".
